buildspec.yml

```diff
@@ -0,0 +1,35 @@
+---
+version: 0.2
+
+env:
+  variables:
+    JQ_INSTALL: https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64
+
+phases:
+  install:
+    commands:
+      - cd tests/
+      - curl -L "${JQ_INSTALL}" -o jq && chmod +x jq                        # install jq
+      - terraform_version=$(curl -s https://checkpoint-api.hashicorp.com/v1/check/terraform | ./jq -r -M '.current_version')
+      - terraform_url="${TERRAFORM_URL:-https://releases.hashicorp.com/terraform/${terraform_version}/terraform_${terraform_version}_linux_amd64.zip}"
+      - curl -L "${terraform_url}" -o tf.zip && unzip tf.zip                    # install terraform
+      - mv terraform /usr/local/bin/ && terraform -version
+  pre_build:
+    commands:
+      - export TF_VAR_codebuild_id="${CODEBUILD_BUILD_ID}"
+      - terraform init -no-color -input=false                                 # init terraform
+      - terraform plan -no-color -out=tfplan -input=false                     # plan terraform
+  build:
+    commands:
+      - terraform apply -no-color -input=false tfplan                        # apply terraform
+  post_build:
+    commands:
+      - ARTIFACT_PATH="${TF_VAR_s3_bucket}/$(terraform output build_date_ymd)/$(terraform output build_date_hm)_$(terraform output build_id)"
```

Comment:
The original rationale for this approach in terrafirm was that we were uploading log files from _within_ the instance, and we wanted the codebuild files to end up in the same bucket key/prefix. We're not doing anything within the instances here, so I think we can instead just lean on the native codebuild "artifact" collection. All the ARTIFACT_PATH/LOCATION and `aws s3 cp` commands can then be replaced with a buildspec `artifacts` definition.

Only the "destroy" logic would remain...

Reply:
By doing this, we also will no longer need the build_date locals/outputs....

Reply:
rgr, removed `aws s3 cp` commands.